File: example/src/main/java/autodagger/example/FifthActivity.java
package autodagger.example;

import android.app.Activity;
import android.os.Bundle;

/**
 * Showcase: @AutoComponent from annotation
 *
 * @author Lukasz Piliszczuk - lukasz.pili@gmail.com
 */
@Deprecated
@StandardActivityComponent2
@DaggerScope(FifthActivity.class)
public class FifthActivity extends Activity {

    private FifthActivityComponent component;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        component = DaggerFifthActivityComponent.builder()
                .myAppComponent(((MyApp) getApplication()).getComponent())
                .build();
        component.inject(this);
    }
}

File: example/src/main/kotlin/autodagger/example/fourth/FifthActivity.kt
package autodagger.example.fourth

import android.app.Activity
import android.os.Bundle
import autodagger.example.DaggerScope
import autodagger.example.KotlinExampleApplication

/**
 * Showcase: @AutoComponent from annotation
 */
@StandardActivityComponent2
@DaggerScope(FifthActivity::class)
class FifthActivity : Activity() {

    private val component: FifthActivityComponent by lazy {
        DaggerFifthActivityComponent.builder()
            .kotlinExampleApplicationComponent((application as KotlinExampleApplication).component)
            .build()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        component.inject(this)
    }
}